src/java/com/google/cloud/bigquery/dwhassessment/hooks/logger/EventRecordConstructor.java

```diff
@@ -239,20 +244,32 @@ private static String getRequestUser(HookContext hookContext) {
     return requestUser == null ? hookContext.getUgi().getUserName() : requestUser;
   }
 
-  private static ExecutionMode getExecutionMode(List<ExecDriver> mrTasks, List<TezTask> tezTasks) {
-    if (tezTasks.size() > 0) {
+  private static ExecutionMode getExecutionMode(QueryPlan plan) {
+    List<ExecDriver> mrTasks = Utilities.getMRTasks(plan.getRootTasks());
```

Comment:
Optional: If `tezTasks` is not empty, it's not necessary to compute `mrTasks`. I would suggest to move `mrTasks` variable definition below the `if (!tezTasks.isEmpty()) { ... }` instruction.

Similarly for `sparkTasks` and `ddlTasks`.

Reply:
Done